src/test/groovy/graphql/execution/AsynchronousExecutionStrategyTest.groovy

```diff
@@ -0,0 +1,130 @@
+                timer.schedule({_-> future.complete([field:"grandValue"]) },50)
+                return future
+        }
+
+        DataFetcher<CompletionStage<String>> parentFetcher = {
+            env ->
+                CompletableFuture<Object> future = new CompletableFuture<>()
+                timer.schedule({_-> future.complete([field:"parentValue"]) },20)
+                return future
+        }
+
+        DataFetcher<CompletionStage<String>> childFetcher = {
+            env ->
+                CompletableFuture<Object> future = new CompletableFuture<>()
+                timer.schedule({_-> future.complete([field:"childValue"]) },10)
+                return future
+        }
+
+        GraphQLObjectType childObjectType = newObject().name("ChildObject").
+                field(newFieldDefinition().name("field").type(GraphQLString)).
+                build();
+
+        GraphQLObjectType parentObjectType = newObject().name("ParentObject").
+                field(newFieldDefinition().name("field").type(GraphQLString)).
+                field(newFieldDefinition().name("child").type(childObjectType).dataFetcher(childFetcher)).
+                build();
+
+        GraphQLObjectType grandObjectType = newObject().name("GrandObject").
+                field(newFieldDefinition().name("field").type(GraphQLString)).
+                field(newFieldDefinition().name("parent").type(parentObjectType).dataFetcher(parentFetcher)).
+                build();
+
+        GraphQLObjectType rootObjectType = newObject().name("Root").
+                field(
+                    newFieldDefinition().name("grand").type(grandObjectType).dataFetcher(grandFetcher)
+                ).build();
+
+        GraphQLSchema schema = GraphQLSchema.newSchema()
+                .query(rootObjectType)
+                .build();
+        when:
+
+        GraphQL graphQL = GraphQL.newGraphQL(schema)
+                .queryExecutionStrategy(new AsynchronousExecutionStrategy())
+                .build();
+
+        String queryString =
+                """
+                {
+                    grand {
+                        field
+                        parent {
+                            field
+                            child {
+                                field
+                            }
+                        }
+                    }
+                }
+                """
+        Map<String,Object> result = ((CompletionStage<Object>) graphQL.execute(queryString).data).toCompletableFuture().get();
+
+        def expected = [
+                grand:[
+                        field: "grandValue",
+                        parent:[
+                                field:"parentValue",
+                                child: [
+                                        field: "childValue"
+                                ]
+                        ]
+                ]
+        ]
+
+        then:
+        assert  result == expected
+    }
+}
```

Comment:
Are we trying to say that the first things to be resolved (grand) is likely to be the last thing to be completed (50ms) and hence it come back in the right order.?